src/frequenz/sdk/timeseries/_formula_engine/_formula_generators/_pv_power_formula.py

```diff
@@ -71,39 +65,35 @@ def generate(self) -> FormulaEngine[Power]:
 
         return builder.build()
 
-    def _get_pv_meters(self) -> abc.Set[int]:
-        component_graph = connection_manager.get().component_graph
+    def _get_pv_power_components(self) -> abc.Set[int]:
+        """Get the component ids of the PV inverters or meters in the component graph.
 
-        pv_inverters = list(
-            comp
-            for comp in component_graph.components()
-            if comp.category == ComponentCategory.INVERTER
-            and comp.type == InverterType.SOLAR
-        )
-        pv_meters: set[int] = set()
-
-        if not pv_inverters:
-            return pv_meters
+        Returns:
+            A set of component ids of the PV inverters or meters in the component graph.
 
-        for pv_inverter in pv_inverters:
-            predecessors = component_graph.predecessors(pv_inverter.component_id)
-            if len(predecessors) != 1:
-                raise FormulaGenerationError(
-                    "Expected exactly one predecessor for PV inverter "
-                    f"{pv_inverter.component_id}, but found {len(predecessors)}."
-                )
-            meter = next(iter(predecessors))
-            if meter.category != ComponentCategory.METER:
-                raise FormulaGenerationError(
-                    f"Expected predecessor of PV inverter {pv_inverter.component_id} "
-                    f"to be a meter, but found {meter.category}."
-                )
-            meter_successors = component_graph.successors(meter.component_id)
-            if len(meter_successors) != 1:
-                raise FormulaGenerationError(
-                    f"Expected exactly one successor for meter {meter.component_id}"
-                    f", connected to PV inverter {pv_inverter.component_id}"
-                    f", but found {len(meter_successors)}."
+        Raises:
+            RuntimeError: if the grid component has no PV inverters or meters as successors.
+        """
+        (
+            component_graph,
+            grid_successors,
+        ) = self._get_graph_and_grid_component_successors()
+
+        if len(grid_successors) == 1:
+            successor = next(iter(grid_successors))
+            if successor.category not in (
+                ComponentCategory.METER,
+                ComponentCategory.INVERTER,
```

Comment:
If there is only one component behind the grid component, that has to be a meter! Otherwise the configuration is screwed. An inverter doesn't has an successor. Thus in that case we are having an inverter directly connected to a grid which implies the whole microgrid is a single inverter. I think we don't have to support such an edge cases in formulas.